qiskit/algorithms/optimizers/umda.py

```diff
@@ -0,0 +1,327 @@
+        )
+
+        self._generation = self._generation[: int(self.ELITE_FACTOR * len(self._generation))]
+        self._generation = np.vstack((self._generation, gen))
+
+    # truncate the generation at alpha percent
+    def _truncation(self):
+        """Selection of the best individuals of the actual generation.
+        Updates the generation by selecting the best individuals.
+        """
+        best_indices = self._evaluations.argsort()[: self._truncation_length]
+        self._generation = self._generation[best_indices, :]
+        self._evaluations = np.take(self._evaluations, best_indices)
+
+    # check each individual of the generation
+    def _check_generation(self, objective_function):
+        """Check the cost of each individual in the cost function implemented by the user."""
+        self._evaluations = np.apply_along_axis(objective_function, 1, self._generation)
+
+    # update the probability vector
+    def _update_vector(self):
+        """From the best individuals update the vector of normal distributions in order to the next
+        generation can sample from it. Update the vector of normal distributions
+        """
+
+        for i in range(self._n_variables):
+            self._vector[0, i], self._vector[1, i] = norm.fit(self._generation[:, i])
+            if self._vector[1, i] < self.STD_BOUND:
+                self._vector[1, i] = self.STD_BOUND
+
+    def minimize(
+        self,
+        fun: Callable[[POINT], float],
+        x0: POINT,
+        jac: Optional[Callable[[POINT], POINT]] = None,
+        bounds: Optional[List[Tuple[float, float]]] = None,
+    ) -> OptimizerResult:
+
+        not_better_count = 0
+        result = OptimizerResult()
+
+        self._n_variables = len(x0)
+        self._best_cost_global = 999999999999
+        self._best_ind_global = 9999999
+        history = []
+        self._evaluations = np.array(0)
+
+        self._vector = self._initialization()
+
+        # initialization of generation
+        self._generation = algorithm_globals.random.normal(
+            self._vector[0, :], self._vector[1, :], [self._size_gen, self._n_variables]
+        )
+
+        for _ in range(self._maxiter):
+            self._check_generation(fun)
+            self._truncation()
+            self._update_vector()
+
+            best_mae_local = min(self._evaluations)
+
+            history.append(best_mae_local)
+            best_ind_local = np.where(self._evaluations == best_mae_local)[0][0]
+            best_ind_local = self._generation[best_ind_local]
+
+            # update the best values ever
+            if best_mae_local < self._best_cost_global:
+                self._best_cost_global = best_mae_local
+                self._best_ind_global = best_ind_local
+                not_better_count = 0
+
+            else:
+                not_better_count += 1
+                if not_better_count == self._dead_iter:
```

Comment:
One question here looking at the logic. Above `self._dead_iter` is set to `int(maxiter/5). So if maxiter is less that 5 then dead_iter will be zero. This code starts out with `not_better_count` being 0. Hence the first if test will check the now incremented value 1 for equality with dead_iter of 0. I am thinking perhaps self._dead_iter should be set to say `min(1, int(maxiter/5)` or this test here should be >=, or both. Honestly >= is probably "safer" than equality as the example I gave would have still broken out in that path.

Reply:
Okay, I think the best way to solve this is to ser >= in the check. I has now been updated. Hope this is all for the review and the rest is fine.